### ISSUE TITLE
Running Titanium Web Proxy in console app on another thread?

### ISSUE THREAD

**Issue by kerray** [justcoding121/Titanium-Web-Proxy#172]
Hi,
I've got a project to create webpage thumbnails using the PhantomJS headless browser, and I need to strip Content-Security-Policy header from our login page in order to login using javascript. 

When I create a standalone console app with Titanium Web Proxy, it works perfectly - the header is removed. 

However I'd like to bundle Titanium in with my thumbnail generator in a single exe, and when I start Titanium inside the app on a new thread, it doesn't work so well - it only goes into the Request handler repeatedly, where it never goes past e.GetResponseBody(), and it doesn't ever go into the Response handler.

I'm not so skilled yet in handling async code, so I wanted to ask, if there is something I need to set, or some other approach I need to try... 

Thanks a lot!

**Comment by miladamirzadeh** [comment]
Would you please post the startup code for the proxy? How do you want to use the proxy in another thread?

**Comment by kerray** [comment]
I'm sorry, I changed my code and I now run Titanium Web Proxy in a separate executable - that works.

The code I had was basically just code from the Titanium Proxy example - I had the Setup code block in a function, and I simply called this function on a new thread
`var newThread = new Thread(TitaniumProxySetup);
newThread.Start();`

The rest of my code utilizes PhantomJS to call our intranet through the proxy.

However, I tried recreating the Proxy startup in separate thread, and now it seems like it works OK. 

So whatever I've done wrong, it's fixed now and we can close this issue.
